core/iwasm/libraries/lib-wasi-threads/stress_test/errorcheck_mutex_stress_test.c

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2023 Amazon.com Inc. or its affiliates. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <pthread.h>
+#include <errno.h>
+#include "mutex_common.h"
+
+int
+main()
+{
+    pthread_mutex_t mutex;
+
+    // Set mutex type to errorcheck. This type provides some additional checks
+    // (for example returns EDEADLK instead of deadlocking in some cases)
```

Comment:
should we have a test that triggers a deadlock and assert the return flag?

Reply:
I did that test and it worked but then I realised it's not a `stress_test` and more like unit/functional test

I can add but I think this basic functional should be already tested or tested somewhere else

Reply:
ok, should we also test different types of mutexes? Especially I'd be interested in a stress test for recursive mutex where we lock the mutex without unlocking multiple times in the same thread.

Reply:
>ok, should we also test different types of mutexes? Especially I'd be interested in a stress test for recursive mutex where we lock the mutex without unlocking multiple times in the same thread.

I think we should (if we don't have such tests already). but I've done that test locally and it was ok (i just locked 5 times and unlocked 5 times)

Reply:
but it's really about wasi-libc testing I think, not even wamr

Reply:
stress tests are both about libc and the runtime. This particular one would allow us to test a particular configuration of assembly instructions related to atomic operations. I presume the assembly code of mutex and recursive mutex operation will be similar, but not sure if exactly the same (just like for the `PTHREAD_MUTEX_ERRORCHECK`).